lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -163,9 +170,19 @@ class DiscoverLoops {
             recordDataflow(connect.getDest(), connect.getSrc());
           })
           .Case<mlir::UnrealizedConversionCastOp>([&](auto) {
-            // Assume external dialects mixed into FIRRTL do not close cycles.
+            // Casts are cast-like regardless of source
+            auto res = op->getResult(0);
+            for (auto src : op->getOperands())
+              recordDataflow(res, src);
           })
           .Default([&](Operation *op) {
+            // Non FIRRTL ops are not checked
+            auto *dialect = op->getDialect();
+            llvm::errs() << dialect << " " << firrtlDialect << " "
+                         << chirrtlDialect << "\n";
+            if (!((firrtlDialect && dialect == firrtlDialect) ||
+                  (chirrtlDialect && dialect == chirrtlDialect)))
+              return;
```

Comment:
Should this be an error or warning? Or at least some message alongside the print, like "comb loop detection ignored <dialect> dialect op; only FIRRTL and CHIRRTL dialect ops are considered". Should this print a message at all?

Reply:
Can we use `isa<CHIRRTLDialect,FIRRTLDialect>(op->getDialect())` instead? I'm not sure the implications for it not being loaded (?).

If there's a concern there, probably would be good to audit this sort of check elsewhere to have the same properties, happy to help there.

Reply:
I think this is a debug print -- if we do have a warning (probably not?) we should surely be careful to avoid printing it a bajillion times -- maybe once per dialect or op or something at best.  Anyway.

Reply:
In the test case, circt-opt hadn't loaded chirrtl, so maybe?  would have to try.

Reply:
It shouldn't be something which appears on the main firrtl compilation path, so I don't know that it's helpful.  It's only going to be introduced by tools intending (presumably) to introduce it, so why warn?

Reply:
I think there's agreement no warning needed here :+1:.

Reply:
(maybe not :) -- don't feel strongly other than if we are gonna limit it which looks like yep)